picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java

```diff
@@ -93,14 +95,12 @@ static void calculateInSampleSize(int reqWidth, int reqHeight, int width, int he
    * about the supplied request in order to do the decoding efficiently (such as through leveraging
    * {@code inSampleSize}).
    */
-  static Bitmap decodeStream(Source source, Request request) throws IOException {
-    BufferedSource bufferedSource = Okio.buffer(source);
-
+  static Bitmap decodeStream(BufferedSource source, Request request) throws IOException {
```

Comment:
Why did you make this change? Since we're using a Source to wrap the incoming source it makes this buffering layer mostly useless. Why not wrap the underlying Source as close as possible to the original before applying any buffering layer?